funnel/metadata.py

```diff
@@ -0,0 +1,37 @@
+import abc
+import typing
+
+import pandas as pd
+import pydantic
+
+
+@pydantic.dataclasses.dataclass
+class BaseMetadataStore(abc.ABC):
+    readonly: bool = False
```

Comment:
How would `readonly` be used here?

Reply:
Exactly.... For now, the in-memory should suffice for testing purposes...